templates/mongodb-node.template

```diff
@@ -111,6 +116,7 @@
             "Type": "String",
             "Default": "3.4",
             "AllowedValues": [
+                "3.6",
```

Comment:
I was also trying to use version 3.6 as well, but it broke for me when trying to make the instance with 3 nodes. This is because MongoDB version 3.6 changed the way they bind a port to localhost. Only clients running on the same machine can connect to localhost. Unlike in 3.4 where an instance will bind itself to localhost by default.
https://docs.mongodb.com/manual/release-notes/3.6-compatibility/#localhost-binding-compatibility-changes

This effectively breaks the orchestrator.sh script being run within PrimaryNode0 from even contacting itself through it's own IP address to connect to the mongo shell.

This can be fixed by adding `echo "  bindIpAll: true" >> mongod.conf` to the init_replica.sh where the mongod.conf file is being built.

I noticed there was also a `enable_all_listen` function in the init_replica script which actually doesn't even appear to be doing anything.

Reply:
I'll try to create a PR for it sometime with what I've been able to learn. Also, I'd like to add t2 instance types to the template as well.

Reply:
Additionally, you'd probably want to allow for [`templates/mongodb-master.template`](https://github.com/aws-quickstart/quickstart-mongodb/blob/master/templates/mongodb-master.template#L322) to allow for version `3.6`.

Reply:
@Camo13ammo This is great, thank you for investigating this. PR is welcome, please be sure to add a comment or message in the commit that the PR is covered by Apache License 2.0.

Reply:
The orchestrator expects the mongo instance to be open to all ips , the "enable_all_listen" seems to be commenting it out any line with "bindIp" to achieve desired behavior, but in mongodb 3.6 when no bindIp is present, it will bind to 127.0.0.1, actually blocking external connections. That's what breaks the whole thing.

if we use `bindIp: 0.0.0.0`, it will reach desired behavior on MongoDB 3.2, 3.4 & 3.6, opposed to `bindIpAll: true` as that is a 3.6 only feature.
I haven't tested what would happen to mongo 3.2, 3.4 when mongo.conf has a flag it does not recognize.

As for limiting access to the cluster, that is done via SecurityGroups, as per-design.

as for `enable_all_listen` function, I too don't understand its purpose, maybe its some old function that never got deleted, I can only think that function to be useful is if we are upgrading a mongodb instance, and we need to make sure its listening to all ips(because thats by design)
Just we can update the sed@
https://github.com/aws-quickstart/quickstart-mongodb/blob/0dc4a940c152e4af63a5a251eade0f02a5a43000/scripts/init_replica.sh#L110
`sed -e '/bindIp/s/^/  bindIp: 0.0.0.0 #/' -i ${f}`